### PR TITLE
ICU-21413 Update laodict.txt per https://github.com/rober42539/lao-dictionary; add tests

### PR DESCRIPTION
**Pull request by pedberg-icu**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21413
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

This is a replacement for [PR-1482](https://github.com/unicode-org/icu/pull/1482) which had a few issues (needed rebasing onto current main; had a duplicate entry for ອັງກິດ which caused build errors; needed tests, and needed to update the java data too). It basically copies the icu4c/source/data/brkitr/dictionaries/laodict.txt file from the earlier PR (roughly tripling the number of entries in the file), removes the duplicate entry, and slightly adjusts the header. In addition it updates the main LICENSE file, updates the ICU4J data as well, and adds basic tests for both C and J.

### CODE REVIEW

File: icu4c/LICENSE [hunk 1]
```diff
@@ -281,31 +281,29 @@ property of their respective owners.
...
+ # Copyright (C) 2016 and later: Unicode, Inc. and others.
+ # License & terms of use: http://www.unicode.org/copyright.html
+ # Copyright (c) 2015 International Business Machines Corporation
+ # and others. All Rights Reserved.
  #
- # Project: https://github.com/veer66/lao-dictionary
- # Dictionary: https://github.com/veer66/lao-dictionary/blob/master/Lao-Dictionary.txt
- # License: https://github.com/veer66/lao-dictionary/blob/master/Lao-Dictionary-LICENSE.txt
- #              (copied below)
+ # Project: https://github.com/rober42539/lao-dictionary
```
Comment (srl295) [inline icu4c/LICENSE:289]: i assume this is the same dictionary?

**Review by srl295 — APPROVED** [review]
lgtm assuming the data is ok and tests pass

File: icu4c/LICENSE (hunk 1 above)
Comment (pedberg-icu) [reply icu4c/LICENSE:289]: Yes, it has been at various places at various times, it was also previously at
http://lao-dictionary.googlecode.com/git/Lao-Dictionary.txt

Comment (rober42539) [reply icu4c/LICENSE:289]: Yeah, I recenymovednit to GitHub, and am now trying to maintain it there. I'm working on a project to add technical vocabulary to it as well. Hopefully I will have some additional data once I can collect it and get it checked.

**Pull request merged**